lua/precognition/init.lua

```diff
@@ -207,14 +93,13 @@ local function build_virt_line(marks, line_len)
     return virt_line
 end
 
----@param buf integer == bufnr
 ---@return Precognition.GutterHints
-local function build_gutter_hints(buf)
+local function build_gutter_hints()
     local gutter_hints = {
-        ["G"] = vim.api.nvim_buf_line_count(buf),
-        ["gg"] = 1,
-        ["{"] = vim.fn.search("^\\s*$", "bn"),
-        ["}"] = vim.fn.search("^\\s*$", "n"),
+        ["G"] = vm.file_end(vim.api.nvim_get_current_buf()),
+        ["gg"] = vm.file_start(),
+        ["{"] = vm.prev_paragraph_line(vim.api.nvim_get_current_buf()),
+        ["}"] = vm.next_paragraph_line(vim.api.nvim_get_current_buf()),
```

Comment:
A lot of calls just take `curbuf`... maybe we should just assume that and not pass bufnr as arguments to vertical motions?

Reply:
Agree

However, I passed it as an optional arg with a default of current, its easier to test that way